### DISCUSSION TITLE
How do I get a bottom border on the selected `NavigationMenuItem`?

### DISCUSSION THREAD

**Discussion by nareshbhatia** [shadcn-ui/ui#2883] [Q&A] [answered]
I am trying to get a bottom border on the selected `NavigationMenuItem`. Something like this:

![image](https://github.com/shadcn-ui/ui/assets/1220198/e4fd90b8-d7f6-4c3e-bd65-715226e3c6f1)

I thought that the `NavigationMenuIndicator` component might do this. However, I couldn't find any examples. I added it to my navigation menu, but it shows nothing.

```tsx
<NavigationMenu>
  <NavigationMenuList>
    <NavigationMenuItem>...</NavigationMenuItem>
    <NavigationMenuItem>...</NavigationMenuItem>
    <NavigationMenuItem>...</NavigationMenuItem>
    <NavigationMenuIndicator />
  </NavigationMenuList>
</NavigationMenu>
```

![image](https://github.com/shadcn-ui/ui/assets/1220198/9e2d12cd-4b00-4f5c-b112-6cd6cc62c2ba)

What is the purpose of `NavigationMenuIndicator` and can I use it to show a bottom border on the selected item?

**Comment by nareshbhatia** [comment]
Hi @shadcn, can you please help me understand what is `NavigationMenuIndicator`? The examples don't cover it (even on the radix-ui side). I tried adding it to your example, but nothing happens.

**Reply by pablotanner** [reply]
I assume it's for the indicator arrow in this image

![image](https://github.com/shadcn-ui/ui/assets/15524263/dfaba637-0f7e-4f84-b674-04bf40ef937b)

**Comment by pablotanner** [comment, marked as answer]
I got the bottom border to work by adding the border to the NavigationMenuTriggers and hiding/showing their color depending on whether the NavigationMenuItem is open or not.
Here's the code, let me know if it works for you!

```javascript
        <NavigationMenu>
                <NavigationMenuList>
                    <NavigationMenuItem>
                        <NavigationMenuTrigger
                            className="border-b-2 border-b-transparent rounded-b-none data-[state='open']:border-black"
                        >
                            First
                        </NavigationMenuTrigger>
                        <NavigationMenuContent>
                            <div className="w-96 h-32 bg-white">
                                Content First
                            </div>
                        </NavigationMenuContent>
                    </NavigationMenuItem>

                    <NavigationMenuItem>
                        <NavigationMenuTrigger
                            className="border-b-2 border-b-transparent rounded-b-none data-[state='open']:border-black"
                        >
                            Second
                        </NavigationMenuTrigger>
                        <NavigationMenuContent>
                            <div className="w-96 h-32 bg-white">
                                Content Second
                            </div>
                        </NavigationMenuContent>
                    </NavigationMenuItem>

                    <NavigationMenuItem>
                        <NavigationMenuTrigger
                            className="border-b-2 border-b-transparent rounded-b-none data-[state='open']:border-black"
                        >
                            Third
                        </NavigationMenuTrigger>
                        <NavigationMenuContent >
                            <div className="w-96 h-32 bg-white">
                                Content Third
                            </div>
                        </NavigationMenuContent>
                    </NavigationMenuItem>
                    <NavigationMenuIndicator />
                </NavigationMenuList>
            </NavigationMenu>
```

**Reply by nareshbhatia** [reply]
Thank you so much, @pablotanner. Your solution gave me an idea of how to implement my use case. In my case, I want the underline when the application path matches the link url. I was able to do it with a change to the `data-active` attribute like [this](https://github.com/nareshbhatia/shadcn-demo/blob/main/src/components/ui/navigation-menu.tsx#L82). The usage example is shown [here](https://github.com/nareshbhatia/shadcn-demo/blob/main/src/app/components/navigation-menu/page.tsx#L73-L75).

**Reply by pablotanner** [reply]
Glad I was able to help!
